src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -448,4 +487,89 @@ end
 
 end probability_measure
 
+section convergence_implies_limsup_closed_le
+
+lemma measure_of_cont_bdd_of_tendsto_filter_indicator {ι : Type*} {L : filter ι}
+  [L.is_countably_generated] [topological_space α] [opens_measurable_space α]
+  (μ : finite_measure α) {c : ℝ≥0} {E : set α} (E_mble : measurable_set E)
+  (fs : ι → (α →ᵇ ℝ≥0)) (fs_bdd : ∀ᶠ i in L, ∀ᵐ (a : α) ∂(μ : measure α), fs i a ≤ c)
+  (fs_lim : ∀ᵐ (a : α) ∂(μ : measure α),
+            tendsto (λ (i : ι), (coe_fn : (α →ᵇ ℝ≥0) → (α → ℝ≥0)) (fs i) a) L
+                    (𝓝 (indicator E (λ x, (1 : ℝ≥0)) a))) :
+  tendsto (λ n, lintegral (μ : measure α) (λ a, fs n a)) L (𝓝 ((μ : measure α) E)) :=
+begin
+  convert finite_measure.tendsto_lintegral_nn_filter_of_le_const μ fs_bdd fs_lim,
+  have aux : ∀ a, indicator E (λ x, (1 : ℝ≥0∞)) a = ↑(indicator E (λ x, (1 : ℝ≥0)) a),
+  from λ a, by simp only [ennreal.coe_indicator, ennreal.coe_one],
+  simp_rw [←aux, lintegral_indicator _ E_mble],
+  simp only [lintegral_one, measure.restrict_apply, measurable_set.univ, univ_inter],
+end
+
+lemma measure_of_cont_bdd_of_tendsto_indicator
+  [topological_space α] [opens_measurable_space α]
+  (μ : finite_measure α) {c : ℝ≥0} {E : set α} (E_mble : measurable_set E)
+  (fs : ℕ → (α →ᵇ ℝ≥0)) (fs_bdd : ∀ n a, fs n a ≤ c)
+  (fs_lim : tendsto (λ (n : ℕ), (coe_fn : (α →ᵇ ℝ≥0) → (α → ℝ≥0)) (fs n))
+            at_top (𝓝 (indicator E (λ x, (1 : ℝ≥0))))) :
+  tendsto (λ n, lintegral (μ : measure α) (λ a, fs n a)) at_top (𝓝 ((μ : measure α) E)) :=
+begin
+  have fs_lim' : ∀ a, tendsto (λ (n : ℕ), (fs n a : ℝ≥0))
+                 at_top (𝓝 (indicator E (λ x, (1 : ℝ≥0)) a)),
+  by { rw tendsto_pi_nhds at fs_lim, exact λ a, fs_lim a, },
+  apply measure_of_cont_bdd_of_tendsto_filter_indicator μ E_mble fs
+      (eventually_of_forall (λ n, eventually_of_forall (fs_bdd n))) (eventually_of_forall fs_lim'),
+end
+
+lemma tendsto_lintegral_thickened_indicator_of_is_closed
+  {α : Type*} [measurable_space α] [pseudo_emetric_space α] [opens_measurable_space α]
+  (μ : finite_measure α) {F : set α} (F_closed : is_closed F) {δs : ℕ → ℝ}
+  (δs_pos : ∀ n, 0 < δs n) (δs_lim : tendsto δs at_top (𝓝 0)) :
+  tendsto (λ n, lintegral (μ : measure α) (λ a, (thickened_indicator (δs_pos n) F a : ℝ≥0∞)))
+          at_top (𝓝 ((μ : measure α) F)) :=
+begin
+  apply measure_of_cont_bdd_of_tendsto_indicator μ F_closed.measurable_set
+          (λ n, thickened_indicator (δs_pos n) F)
+          (λ n a, thickened_indicator_le_one (δs_pos n) F a),
+  have key := thickened_indicator_tendsto_indicator_closure δs_pos δs_lim F,
+  rwa F_closed.closure_eq at key,
+end
+
+lemma finite_measure.limsup_measure_closed_le_of_tendsto
+  {α ι : Type*} {L : filter ι} [ne_bot L]
```

Comment:
You shouldn't need `ne_bot` here: if I'm not mistaken, when the filter is bot then the limsup is 0, so the statement is still true.

Reply:
Thank you for pointing this out! This will for sure make the result more convenient to use. (The proof now splits to cases, though, but the `L = ⊥` case is one `simp`.)